docs/diagrams/FindCommandSequenceDiagram.puml

```diff
@@ -0,0 +1,116 @@
+participant "predicate:Predicate" as Predicate MODEL_COLOR
+participant "model:Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute
+activate LogicManager LOGIC_COLOR
+
+LogicManager -> AddressBookParser : parseCommand(commandText)
+activate AddressBookParser LOGIC_COLOR
+
+create FindCommandParser
+AddressBookParser -> FindCommandParser
+activate FindCommandParser LOGIC_COLOR
+FindCommandParser --> AddressBookParser
+deactivate FindCommandParser
+
+AddressBookParser -> FindCommandParser : parse(arguments)
+activate FindCommandParser LOGIC_COLOR
+
+FindCommandParser -> ArgumentTokenizer ++ LOGIC_COLOR: tokenize(argsString, prefixes)
+ArgumentTokenizer --> FindCommandParser -- LOGIC_COLOR: argMultimap
+
+ref over FindCommandParser: Parsing inputted field
+
+create Predicate
+FindCommandParser -> Predicate
+activate Predicate MODEL_COLOR
+Predicate --> FindCommandParser
+deactivate Predicate
+
+create FindCommand
+FindCommandParser -> FindCommand
+activate FindCommand LOGIC_COLOR
+FindCommand --> FindCommandParser
+deactivate FindCommand
+
+FindCommandParser --> AddressBookParser -- LOGIC_COLOR: FindCommand
+AddressBookParser --> LogicManager -- LOGIC_COLOR: FindCommand
+
+LogicManager -> FindCommand ++ LOGIC_COLOR: execute(model)
+FindCommand -> Model ++ MODEL_COLOR: updateFilteredPersonList(predicate)
+deactivate Model
+
+create CommandResult
+FindCommand -> CommandResult
+activate CommandResult LOGIC_COLOR
+CommandResult --> FindCommand
+deactivate CommandResult
+
+FindCommand --> LogicManager -- LOGIC_COLOR: commandResult
+[<--LogicManager --: commandResult
+deactivate LogicManager
+
+@enduml
+
+@startuml
+!include style.puml
+
+box "Logic" LOGIC_COLOR_T1
+participant ":FindCommandParser" as FindCommandParser LOGIC_COLOR
+participant ":ParserUtil" as ParserUtil LOGIC_COLOR
+end box
+
+box "Model" MODEL_COLOR_T1
+participant "studentId:StudentId" as StudentId MODEL_COLOR
+participant "name:Name" as Name MODEL_COLOR
+participant "moduleCode:ModuleCode" as ModuleCode MODEL_COLOR
+end box
+
+mainframe sd Parsing inputted field
+
+alt user inputs studentId
+    FindCommandParser -> ParserUtil ++ LOGIC_COLOR: parseStudentId(studentId)
+    create StudentId
+    ParserUtil -> StudentId
+    activate StudentId MODEL_COLOR
+    StudentId --> ParserUtil
+    deactivate StudentId
+    ParserUtil --> FindCommandParser -- MODEL_COLOR: studentId
+end
+
+alt user inputs name
+    FindCommandParser -> ParserUtil ++ LOGIC_COLOR: parseName(name)
+    create Name
+    ParserUtil -> Name
+    activate Name MODEL_COLOR
+    Name --> ParserUtil
+    deactivate Name
+    ParserUtil --> FindCommandParser -- MODEL_COLOR: name
+end
+
+alt user inputs moduleCode
+    FindCommandParser -> ParserUtil ++ LOGIC_COLOR: parseModuleCode(moduleCode)
+    create ModuleCode
+    ParserUtil -> ModuleCode
+    activate ModuleCode MODEL_COLOR
+    ModuleCode --> ParserUtil
+    deactivate ModuleCode
+    ParserUtil --> FindCommandParser -- LOGIC_COLOR: moduleCode
+end
```

Comment:
I'm not too sure if this is right way to represent alternative paths. Taking reference from the sequence diagram for `Task`:
![TaskCommandSequenceDiagram-3](https://user-images.githubusercontent.com/46317008/159557054-176a3955-52f0-4279-b3b7-fa7932b4a291.png)

If I'm not wrong, it's a matter of changing the `alt` for the other branches to `else`.

```suggestion
alt user inputs studentId
    FindCommandParser -> ParserUtil ++ LOGIC_COLOR: parseStudentId(studentId)
    create StudentId
    ParserUtil -> StudentId
    activate StudentId MODEL_COLOR
    StudentId --> ParserUtil
    deactivate StudentId
    ParserUtil --> FindCommandParser -- MODEL_COLOR: studentId
end

else user inputs name
    FindCommandParser -> ParserUtil ++ LOGIC_COLOR: parseName(name)
    create Name
    ParserUtil -> Name
    activate Name MODEL_COLOR
    Name --> ParserUtil
    deactivate Name
    ParserUtil --> FindCommandParser -- MODEL_COLOR: name
end

else user inputs moduleCode
    FindCommandParser -> ParserUtil ++ LOGIC_COLOR: parseModuleCode(moduleCode)
    create ModuleCode
    ParserUtil -> ModuleCode
    activate ModuleCode MODEL_COLOR
    ModuleCode --> ParserUtil
    deactivate ModuleCode
    ParserUtil --> FindCommandParser -- LOGIC_COLOR: moduleCode
end
```
Ref image from plantuml:
![image](https://user-images.githubusercontent.com/46317008/159557427-ff805ab9-245c-4ad6-b76e-5e93b653b3f8.png)

Reply:
Hi @brelkh, thanks for the review! You're right about using `else` instead of `alt` for the other branches. Regarding your suggested code, the only changes are that there should only be one `end` at the end of the alt clause, rather than at the end of every `alt` and `else` clause. I've updated my code along with the diagram.